src/Stoa.ts

```diff
@@ -381,59 +381,109 @@ class Stoa extends WebService
+     *      The "inbound" is an increased transaction of funds at the address.
+     *      The "outbound" is a transaction with reduced funds at the address.
+     *      Users can select only "inbound", "outbound".
+     *      The "freeze", "payload" are always included.
+     *      default is "inbound,outbound,freeze,payload"
+     * The parameter `beginDate` is the start date of the range of dates to look up.
+     * The parameter `endDate` is the end date of the range of dates to look up.
+     * The parameter `peer` is used when users want to look up only specific
+     *      address of their counterparts.
+     *      Peer is the withdrawal address in the inbound transaction and
+     *      a deposit address in the outbound transaction
      * Returns a set of transactions history of the addresses.
+     * ```
      */
     private getWalletTransactionsHistory (req: express.Request, res: express.Response)
     {
-        let params_addresses: string = String(req.params.addresses);
+        let address: string = String(req.params.address);
 
-        logger.http(`GET /wallet/transactions/history/${params_addresses}}`);
+        logger.http(`GET /wallet/transactions/history/${address}}`);
 
-        let page_size = (req.query.pageSize !== undefined)
-            ? Number(req.query.pageSize.toString())
-            : 10;
+        if ((req.query.beginDate !== undefined) &&
+            !Utils.isPositiveInteger(req.query.beginDate.toString()))
+        {
+            res.status(400).send(`Invalid value for parameter 'beginDate': ${req.query.beginDate.toString()}`);
+            return;
+        }
 
-        let page = (req.query.page !== undefined)
-            ? Number(req.query.page.toString())
-            : 1;
+        if ((req.query.endDate !== undefined) &&
+            !Utils.isPositiveInteger(req.query.endDate.toString()))
+        {
+            res.status(400).send(`Invalid value for parameter 'endDate': ${req.query.endDate.toString()}`);
+            return;
+        }
 
-        if (page_size <= 0)
+        if ((req.query.pageSize !== undefined) &&
+            !Utils.isPositiveInteger(req.query.pageSize.toString()))
         {
-            res.status(400).send(`Positive pageSize expected: ${page_size}`);
+            res.status(400).send(`Invalid value for parameter 'pageSize': ${req.query.pageSize.toString()}`);
             return;
         }
 
-        if (page_size > 30)
+        if ((req.query.page !== undefined) &&
+            !Utils.isPositiveInteger(req.query.page.toString()))
         {
-            res.status(400).send(`Page size cannot be a number greater than 30: ${page_size}`);
+            res.status(400).send(`Invalid value for parameter 'page': ${req.query.page.toString()}`);
             return;
         }
 
-        if (page <= 0)
+        let page_size = (req.query.pageSize !== undefined)
+            ? Number(req.query.pageSize.toString())
+            : 10;
```

Comment:
You can handle it together on line 430.

Reply:
When the two are combined, it becomes like this and the structure is not easy to understand, so we separated them.
And the top code is the verification code for when an error occurs when converted to a number.
I think it's better to separate the verification code from the actual code
``` TypeScript
        let page_size: number;
        if (req.query.pageSize !== undefined) {
            if (!Utils.isPositiveInteger(req.query.pageSize.toString())) {
                res.status(400).send(`Invalid value for parameter 'pageSize': ${req.query.pageSize.toString()}`);
                return;
            }
            page_size = Number(req.query.pageSize.toString())
        }
        else page_size = 10
```

Reply:
There will now be more parameter verification routines.
I want to unify the validation routine.
But now we can go this way.